### PR TITLE
documents: configure OAI sets

### LINKED ISSUE #311: OAI-PMH Automatic creation sets for organisation

### PR DESCRIPTION
**Pull request by sebastiendeleze**
* Creates an OAI set when an organisation is created.
* Configures OAI ID prefix.
* Updates JSON schema and elasticsearch mapping for documents to store the `_oai` property.
* Updates the `_oai` property of record when a document is created or updated, accordingly to his organisation.
* Serializes the `_oai` property when document is sent to API.
* Closes #311.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

### CODE REVIEW

File: sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json [hunk 1]
```diff
@@ -113,6 +113,35 @@
...
+            "minLength": 1
+          }
+        },
+        "updated": {
+          "description": "Last OAI-PMH set update (ISO8601 formatted timestamp)",
+          "type": "string",
+          "minLength": 1
+        }
+      },
+      "required": [
```
Comment (zannkukai) [inline sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json:140]: For me, `updated` property should be required. When a document is created, then the `updated` field should be equal to creation date

**Review by zannkukai — APPROVED** [review]
Code seems work, so I approved.
How will you manage the "deleted record" problem ? With OAI, you should provide an "updated record list" but also a "deleted record list" in another set. In all systems that I knows the "delete set" is often problematic because, I you physically delete the record from invenio, you can't know that's the record was deleted. Am I wrong ? maybe Invenio store an information about the pid status in the DB (with operation date) ?

File: sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json (hunk 1 above)
Comment (sebastiendeleze) [reply sonar/modules/documents/jsonschemas/documents/document-v1.0.0_src.json:140]: Absolutely right, corrected!

**Comment by sebastiendeleze** [comment]
Thank you for your review and your comment. I was not aware of the existence of deleted sets. In fact, when a record is deleted, the PID is kept but marked as deleted:

Capture d’écran 2020-09-29 à 16 44 27

But the metadata are trashed... I have to discuss with @mmo about how handle this type of sets.

**Comment by pronguen** [comment]
I tested it on sonardev.test.rero.ch. The request `https://sonardev.test.rero.ch/oai2d?verb=ListSets` does not provide any sets. How can we test it?

**Review by pronguen — APPROVED** [review]
Ok, but I didn't test the harvest/syncho feature!

**Pull request merged**